src/locale/localed-util.c

```diff
@@ -353,6 +373,25 @@ int vconsole_write_data(Context *c) {
         if (r < 0)
                 return r;
 
+        xc = context_get_x11_context(c);
+        if (xc) {
+                r = strv_env_assign(&l, "XKB_LAYOUT", empty_to_null(xc->layout));
+                if (r < 0)
+                        return r;
+
+                r = strv_env_assign(&l, "XKB_MODEL", empty_to_null(xc->model));
+                if (r < 0)
+                        return r;
+
+                r = strv_env_assign(&l, "XKB_VARIANT", empty_to_null(xc->variant));
+                if (r < 0)
+                        return r;
+
+                r = strv_env_assign(&l, "XKB_OPTIONS", empty_to_null(xc->options));
```

Comment:
one of those days we might want to add strv_env_assign_many() or so with a varargs parm